config-store-api/src/main/java/io/stargate/config/store/ConfigStore.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.config.store;
+
+import java.util.Map;
+
+public interface ConfigStore {
+
+  /**
+   * It retrieves the generic {@code Map<String, Object>} for a given extension name.
+   *
+   * <p>If the caller will try to look up the config for an extension that does not have any
+   * setting, it will throw the {@link MissingExtensionSettingsException}. This is a {@link
```

Comment:
Why throw instead of returning an empty `Map`? Is it really an error for an extension to have no config settings defined?

Since the whole config is dynamic, settings may appear later... Would it be more appropriate to just log an error message in this case?

Reply:
From the `ConfigStore` perspective, this is an abnormal situation that the client asks for the config, but the config does not exist. It is up to the caller to decide what to do with that fact. By being explicit in the API, we are allowing clients to provide any missing setting handling. If we would hide it and log that fact, callers will not be able to decide anymore.

Reply:
+1 to explicit API, but from the practical perspective, can we be sure that config and jars are always in sync during deployment?

Reply:
The `stargate-config.yaml` will be mounted in a pod before the deployment of the stargate, see:
https://github.com/riptano/cloud-ondemand/pull/2339
So I think it guarantees the happens-before and files will be in sync.
When the file will be refreshed we would need to only load it again.